salt/_modules/caasp_cri.py

```diff
@@ -0,0 +1,389 @@
+
+    if cri == 'docker':
+        return 'docker'
+    elif cri == 'crio':
+        return 'crio'
+    else:
+        raise InvalidConfigError(
+                'Uknown CRI specified inside of pillars: {}'.format(cri))
+
+
+def cri_salt_state_name():
+    '''
+    Return a string holding the name of the salt state that manages the CRI.
+
+    This is used internally by our salt states to DRY them.
+    '''
+    return cri_service_name()
+
+
+def cri_package_name():
+    '''
+    Return a string holding the name of the package providing the CRI
+
+    This is used internally by our salt states to DRY them.
+    '''
+    if 'admin' in __salt__['grains.get']('roles', []):
+        return 'docker'
+
+    cri = __salt__['pillar.get']('cri:name', 'docker').lower()
+
+    if cri == 'docker':
+        return 'docker'
+    elif cri == 'crio':
+        return 'cri-o'
+    else:
+        raise InvalidConfigError(
+                'Uknown CRI specified inside of pillars: {}'.format(cri))
+
+
+def cri_runtime_endpoint():
+    '''
+    Return the path to the socket required by crictl to communicate
+    with the CRI
+    '''
+    if 'admin' in __salt__['grains.get']('roles', []):
+        return '/var/run/dockershim.sock'
+
+    cri = __salt__['pillar.get']('cri:name', 'docker').lower()
+
+    if cri == 'docker':
+        return '/var/run/dockershim.sock'
+    elif cri == 'crio':
+        return '/var/run/crio/crio.sock'
+    else:
+        raise InvalidConfigError(
+                'Uknown CRI specified inside of pillars: {}'.format(cri))
+
+
+def _docker_cp_file_to_container(container_id, source, destination):
+    '''
+    Use ``docker cp`` to copy a file from the host into a running container
+    managed by docker
+    '''
+
+    cmd = 'docker cp {source} {container_id}:{destination}'.format(
+            source=source,
+            container_id=container_id,
+            destination=destination)
+    result = __salt__['cmd.run_all'](cmd,
+                                     output_loglevel='trace',
+                                     python_shell=False)
+
+    if result['retcode'] != 0:
+        return (False, result['stderr'])
+
+    return (True, '')
+
+
+def _crio_cp_file_to_container(container_id, source, destination):
+    '''
+    Use ``podman`` to copy a file from the host into a running container
+    managed by crio.
+    '''
+
+    cmd = 'podman mount {container_id}'.format(
+            container_id=container_id)
+    result = __salt__['cmd.run_all'](cmd,
+                                     output_loglevel='trace',
+                                     python_shell=False)
+
+    if result['retcode'] != 0:
+        return (
+                False,
+                'Error mounting container fs with podman: {}'.format(
+                    result['stderr']))
+
+    try:
+        if os.path.isdir(source):
+            shutil.copytree(source, destination)
+        else:
+            shutil.copy(source, destination)
```

Comment:
Currently, we ignore the returned mount point from `podman mount`. I guess we need to copy to `mountpoint/destination`.

Reply:
Good point, I'll tackle that.